### ISSUE TITLE
Enable automatic night mode when in a tunnel

### ISSUE THREAD

**Issue by cmeeren** [osmandapp/Osmand#2724]
Day mode can be rather bright in tunnels. I suggest adding an option to the Map Mode menu (Day, Night, Sunrise/Sunset and Light sensor) to enable night mode when in a tunnel.

Before you suggest just using the light sensor for this, there are two problems:

1. Not all devices have light sensors (e.g., in-car head units with Android - I have installed OsmAnd there and am testing using it as my main navigation app).
2. The light sensor threshold is not adjustable, and frequently it shifts to night mode even in bright daylight, when e.g. driving through a heavily forested area (where the ambient light is still bright enough that night mode poses a problem).

**Comment by aceman444** [comment]
Yes, when "snap to road" is on, OsmAnd may guess when you are in a tunnel also without a sensor.

**Comment by vshcherb** [comment]
Yes, it can.

**Comment by teatimetom** [comment]
Is there any update?

**Comment by scaidermern** [comment]
This is already possible when setting day/night mode to light sensor (Configure map -> Map mode). However it currently depends on your device whether this works well since the sensitivity is not configurable (see #5001).

**Comment by cmeeren** [comment]
Please read the OP:

> Before you suggest just using the light sensor for this, there are two problems: (...)

**Comment by scaidermern** [comment]
> Please read the OP:
> 
> > Before you suggest just using the light sensor for this, there are two problems: (...)

Sorry, my fault. However for implementing this it would be nice to have a more reliable tunnel detection (#5879).

**Comment by teatimetom** [comment]
Any news?

**Comment by toas-koas** [comment]
Any update?

**Comment by reibax-marcus** [comment]
This would be nice
